web/src/billing/sponsorship.ts

```diff
@@ -99,6 +99,11 @@ export function initialize(): void {
         return;
     }
 
+    if ($(".demo-organization-warning").length > 0) {
+        // Sponsorship form is not present for demo organizations.
+        return;
+    }
```

Comment:
In particular, we'd be able to avoid having to think about and maintain potentially fragile corner case code like this if we just served an error page rather than having a copy of the actual page with everything hidden.